src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCITriggerService.java

```diff
@@ -71,10 +74,15 @@ public class LocalCITriggerService implements ContinuousIntegrationTriggerServic
 
     private final GitService gitService;
 
-    private IQueue<BuildJobQueueItem> queue;
+    private IQueue<BuildJobItemReferenceDTO> queue;
 
     private IMap<String, ZonedDateTime> dockerImageCleanupInfo;
 
+    private IMap<Long, CircularFifoQueue<BuildJobItem>> buildJobItemMap;
+
+    @Value("${artemis.continuous-integration.max-build-job-items:2}")
```

Comment:
My first impression when reading `max-build-job-items` is that this limits concurrency for the whole system, not just per participation. Can you find a better name for this configuration parameter? Maybe `parallel-jobs-per-participation`.